### PR TITLE
Doesn't render the badge if there's no children

### LINKED ISSUE #76: Badge empty text case
If in result of component render we've got empty attribute in DOM now, attribute should not be rendered at all.

So it should not be required too.

### PR DESCRIPTION
**Pull request by tleunen**
fixes #76

### CODE REVIEW

**Comment by faergeek** [comment]
Sorry, it's not what I'm talking about. It's about `text` prop, not `children`.

The point is that we're wrapping some element in `Badge` and we can provide `text`, but according to upstream styles the element is shown only if DOM element has a `data-badge` attribute, so if it's empty (there's no new notifications or something like that) then it doesn't make sense to render it.

**Comment by faergeek** [comment]
Here's a selector `.mdl-badge[data-badge]:after`.

**Comment by tleunen** [comment]
Haha ok ;) Yep, actually I figured out it make no sense if the children was null as well anyway ;)
About the text, I'm not sure.. How would you render an empty badge? (just the red circle without any content)?
I'm not saying there's no fix to be made on react-mdl. But maybe we could render an empty data-badge if text is null? Because if you don't want the badge, you could not render it in your component instead of assuming Badge will not render himself

**Comment by faergeek** [comment]
Let's assume that we have some icon for notifications and we want to show a badge only if there's a notification (mostly like on github, but with numbers), with the current implementation we will wrap component in a badge if there's notifications or don't wrap if there's none. Not so good.

Instead I guess it's assumed by upstream that if we don't want to show badge we just don't set `data-badge` attribute, even empty.

I can make a PR to show the difference.

**Comment by tleunen** [comment]
Ok so you suggest that if `text` is null, `Badge:render` returns the children then?

**Comment by faergeek** [comment]
Not exactly. I'll make a PR to explain better :-)

**Comment by faergeek** [comment]
It throws now when you return `null` from `Badge`.

**Comment by faergeek** [comment]
What if instead of checking for a single element and so on and so forth we just wrap element in span in any case? Try to wrap `Textfield` in `Badge`, for example. It's confusing.

**Comment by tleunen** [comment]
No need to add an extra container for that.
What do you mean when you return `null` from `Badge`?

File: src/Badge.js [hunk 1]
```diff
@@ -5,12 +5,16 @@ class Badge extends React.Component {
...
-        ]).isRequired,
+        ]),
         text: PropTypes.string.isRequired
     }
 
     render() {
         var { children } = this.props;
+
+        // No badge if no children
+        if(!React.Children.count(children)) return null;
```
Comment (faergeek) [inline src/Badge.js]: I'm talking about this line.
It seems like stateless components do not support returning null from render.
Current test suite skips it somehow.

Comment (tleunen) [inline src/Badge.js]: Hmm weird. Do you know if there's already a bug about that on react?
Seems weird the behavior from a stateless component is different in React and in a Shallow renderer.

Comment (faergeek) [inline src/Badge.js]: I guess may be we just don't get stateless components right?)
I didn't search for a bug.

**Pull request merged**